src/classes/Recipe.js

```diff
@@ -9,36 +9,40 @@ class Recipe {
   }
 
   matchIngredients(ingredientData) {
-    const ingredientNames = this.ingredients.map(ingredient => {
-      const ingredientDataNames = ingredientData.filter(element => element.id === ingredient.id);
-      return ingredientDataNames;
+    const relevantIngredients = ingredientData.filter(ingredient => {
+      const recipeIds = this.ingredients.map(ingredient => ingredient.id);
+      return recipeIds.includes(ingredient.id);
     });
-    return ingredientNames.flat();
+    
+    const ingredientsWithAllData = this.ingredients.reduce((acc, cv) => {
+      const combiner = relevantIngredients.forEach(el => {
+        if(el.id === cv.id) {
+          acc.push(
+          {
+            id: el.id,
+            name: el.name,
+            estimatedCostInCents: el.estimatedCostInCents,
+            quantity: cv.quantity
+          });
+        }
+      });
+      return acc;
+    }, []);
+
+    return ingredientsWithAllData;
   }
   
   listIngredients(ingredientData) {
```

Comment:
Unsure if/how we'll need to change these methods/parameters once we're using `fetch` to import data. Putting a pin it as this is unfamiliar territory

Reply:
I agree. I think it might still work the same once the data is parsed with json and stored in a variable..?

Reply:
Ok sounds like a plan for now.